service/src/main/java/com/epam/esm/service/builder/certificate/GiftCertificateBuilder.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.esm.service.builder.certificate;
+
+import com.epam.esm.dao.entity.GiftCertificate;
+import com.epam.esm.service.builder.tag.TagBuilder;
+import com.epam.esm.service.dto.GiftCertificateDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.stream.Collectors;
+
+@Component
+public class GiftCertificateBuilder {
+  private final TagBuilder tagBuilder;
+
+  @Autowired
+  public GiftCertificateBuilder(TagBuilder tagBuilder) {
+    this.tagBuilder = tagBuilder;
+  }
+
+  public GiftCertificate buildNewParameterGiftCertificate(
```

Comment:
I think it can be a good approach to separate this method into small methods like 'addDescription, addPrice'
And this method can be just 
build() {
addDescription(...);
addPrice(...);
...
return certificate;
}

Reply:
Thank you for your idea. I'm sure this approach is more appropriate. (fixed)